### PR TITLE
feat: Restricting visibility of contact attributes to approved accounts

### PR DESCRIPTION
**Pull request by martintomas**
Making sure that contact information attributes for PD endpoints are restricted and only user with approved account can see them.

## Testing instructions

You can trust tests or rswag ui :)

## Tracking

https://vizzuality.atlassian.net/browse/LET-318

### CODE REVIEW

File: backend/app/models/account.rb [hunk 1]
```diff
@@ -29,4 +29,8 @@ def slug_preview
     set_slug unless slug.present?
     slug
   end
+
+  def approved?
+    investor&.approved? || project_developer&.approved?
```
Comment (tsubik) [inline backend/app/models/account.rb:34]: I'm not sure why we have approved on investor/project_developer and not directly on account, maybe that is just a mistake? @agnessa

**Review by tsubik — COMMENTED** [review]
Are there any API specs for logged-in user? I see all snapshots changing those attributes to `null` so those are for public users.

**Comment by martintomas** [comment]
> Are there any API specs for logged-in user? I see all snapshots changing those attributes to `null` so those are for public users.

You should be able to see filled values for example at https://github.com/Vizzuality/heco-invest/blob/feature/restrict-visibility-contacts/backend/spec/fixtures/snapshots/api/v1/accounts-project-developer.json#L27 where response is returned for user with approved account.

File: backend/app/models/account.rb (hunk 1 above)
Comment (martintomas) [reply backend/app/models/account.rb:34]: Yeah, hmm, I agree with you, if `Account` model is used to save data which are same for Investor and Project Developer, then `approved` attributes should be moved to it. If you all of us agree, I can refactor code this way :)

Comment (agnessa) [reply backend/app/models/account.rb:34]: agree

**Review by agnessa — APPROVED** [review]

File: backend/app/models/account.rb (hunk 1 above)
Comment (tsubik) [reply backend/app/models/account.rb:34]: Yes, better to move it. Seems more logical to have account.approved directly there.

**Comment by tsubik** [comment]
@martintomas but this is a different controller, the private/account one. I'm talking about some one extra spec for public controller to for logged-in user. To make sure contact info is returned.

**Comment by martintomas** [comment]
> @martintomas but this is a different controller, the private/account one. I'm talking about some one extra spec for public controller for logged-in user. To make sure contact info is returned.

Aha, okay, I guess there is no one. I though that private/account one is enough, because serializer is still same, but I can add new test to public endpoint too, np

**Pull request merged**